relayer/pkg/chainlink/txm/txm.go

```diff
@@ -250,47 +260,54 @@ func (txm *starktxm) broadcast(ctx context.Context, publicKey *felt.Felt, accoun
 
 	// TODO: consider making this configurable
 	// pad estimate to 250% (add extra because estimate did not include validation)
-	gasConsumed := friEstimate.GasConsumed.BigInt(new(big.Int))
+	gasConsumed := friEstimate.L1GasConsumed.BigInt(new(big.Int))
```

Comment:
in this section it's getting hard to track what values we're updating. but we're essentially updating the resourceBounds for the L1Gas, L2Gas, and L1DataGas. And for each one the MaxAmount and MaxPricePerUnit fields. So 6 fields in total.

Maybe you can create a helper method here so it's less confusing and we can avoid similar bugs uncovered in the future. 

You could create a function like

func updateMaxAmountBounds(maxAmount *U64, estimate *big.Int) {}

and 

func updateMaxPricePerUnit(maxAmount *U128, estimate *big.Int) {}

and the logic would be the same, just multiplying by the pad.

or something like that where we can consolidate some of the logic and make it more clear which values are being updated.